gh-95273: Improve sqlite3 class descriptions (#95379)
python/cpython: Doc/library/sqlite3.rst

.. class:: Connection

>    Each open SQLite database is represented by a connection object.
>    A connection object is created with :func:`sqlite3.connect`.

Reviewer: ```suggestion
   Each open SQLite database is represented by a ``Connection`` object,
   which is created using :func:`sqlite3.connect`.
```

* Refer to the literal object class
* Avoid repeating ourselves ("connection objects" three times in a row)
* Tweak phrasing to improve flow and concision

>    The main purpose of connection objects is creating

Reviewer: ```suggestion
   Their main purpose is creating
```

Avoid repetition and be more succinct

>    :class:`cursors objects <Cursor>`,

Reviewer: ```suggestion
   :class:`Cursor` objects
```

Avoid grammar issues and whole expression being made (unintentionally) literal, and use literal class name

Cursor Objects
--------------

   A ``Cursor`` object represents a `database cursor`_
   which is used to execute SQL statements,
   and manage the context of a fetch operation.

>    Cursors are created with :meth:`Connection.cursor`,

Author: ```suggestion
   Cursors are created using :meth:`Connection.cursor`,
```

   or by using any of the :ref:`connection shortcut methods
   <sqlite3-connection-shortcuts>`.

>    Cursors objects are :term:`iterators <iterator>`,

Author: ```suggestion
   Cursor objects are :term:`iterators <iterator>`,
```

Cursor Objects
--------------

>    A cursor object represents a database cursor,

Reviewer: Consider linking this with a reference summarizing what a database cursor is for the uninitiated (like me), such as

```suggestion
   A ``Cursor`` object represents a
   `database cursor <https://en.wikipedia.org/wiki/Cursor_(databases)>`_,
```

   which is used to execute SQL statements,
   and manage the context of a fetch operation.
   Cursors are created with :meth:`Connection.cursor`,
   or by using any of the :ref:`connection shortcut methods
   <sqlite3-connection-shortcuts>`.

   Cursors objects are :term:`iterators <iterator>`,

>    meaning that if you :meth:`execute` a ``SELECT`` query,

Reviewer: ```suggestion
   meaning that if you :meth:`~Cursor.execute` a ``SELECT`` query,
```

You aren't inside the class here, so the reference is broken (and generates a warning)

.. class:: Row

>    A :class:`Row` instance that serves as a highly optimized

Author: Maybe the original was better.

Reviewer: Yeah, the revised wording doesn't really make much sense—it implies a ``:class:`Row`` _class_ (the title of the section, and thus the implied subject) is a ``:class:`Row`` _instance_, which is not what you intend.

```suggestion
   A :class:`Row` instance serves as a highly optimized
```

   :attr:`~Connection.row_factory` for :class:`Connection` objects.

>    It tries to mimic a :class:`tuple` in most of its features,

Author: Perhaps "Row objects" instead of "It".

Reviewer: As part of the same paragraph and with the reversion above, the subject is unambiguous and saying "Row objects" again would be repetitive, IMO

>    and supports :term:`mapping` access by column name and index, iteration,
>    representation, equality testing and :func:`len`.

Reviewer: ```suggestion
   and supports iteration; :func:`repr`, equality testing, :func:`len`,
   and :term:`mapping` access (by column name and index).
```

* Restructure to avoid syntactic ambiguity
* Use (what I presume is the intended) `repr` instead of "representation", which is much more clear about what it refers to (I've rarely if ever heard `repr` called that)

Author: That's better, but I prefer to leave out the parentheses around column name and index.

>    Two row objects are equal if they have equal columns and equal members.

Reviewer: ```suggestion
   Two row objects compare equal if have equal columns and equal members.
```

* Use slightly clearer and more idiomatic phrasing

Author: No _they_? TIL
